File: C/Sully.c
#include <stdio.h>
#include <stdlib.h>
#define NB 5
#define EXE sprintf(exe, "Sully_%d", i)
#define SRC sprintf(src, "Sully_%d.c", i)
#define CMD sprintf(cmd, "gcc -Wall -Wextra -Werror %1$s -o %2$s && ./%2$s", src, exe)
#define FT() int main(){\
    int i = 5;\
    char *c="#include <stdio.h>%1$c#include <stdlib.h>%1$c#define NB 5%1$c#define EXE sprintf(exe, %2$cSully_%%d%2$c, i)%1$c#define SRC sprintf(src, %2$cSully_%%d.c%2$c, i)%1$c#define CMD sprintf(cmd, %2$cgcc -Wall -Wextra -Werror %%1$s -o %%2$s && ./%%2$s%2$c, src, exe)%1$c#define FT() int main(){%4$c%1$c    int i = %5$d;%4$c%1$c    char *c=%2$c%3$s%2$c;%4$c%1$c    FILE *ptr;%4$c%1$c    char src[1024];SRC;%4$c%1$c    char exe[1024];EXE;%4$c%1$c    char cmd[4096];CMD;%4$c%1$c    if (i<0 || !(ptr = fopen(src,%2$cw+%2$c)))exit(0);%4$c%1$c    fprintf(ptr, c, 10, 34, c, 92, i - 1);%4$c%1$c    fclose(ptr);%4$c%1$c    if (!i)exit(0);%4$c%1$c    system(cmd);}%1$cFT()";\
    FILE *ptr;\
    char src[1024];SRC;\
    char exe[1024];EXE;\
    char cmd[4096];CMD;\
    if (i<0 || !(ptr = fopen(src,"w+")))exit(0);\
    fprintf(ptr, c, 10, 34, c, 92, i - 1);\
    fclose(ptr);\
    if (!i)exit(0);\
    system(cmd);}

FT()

File: GO/Sully.go
package main
import (
        "fmt"
        "os"
        "strings"
        "os/exec"
        )
func main(){
        i := 5
        pgm, _ := os.Executable()
        dec := strings.HasSuffix(strings.ToLower(pgm), "/sully")
        if dec == false {i--}
        c :="package main%[1]cimport (%[1]c        %[2]cfmt%[2]c%[1]c        %[2]cos%[2]c%[1]c        %[2]cstrings%[2]c%[1]c        %[2]cos/exec%[2]c%[1]c        )%[1]cfunc main(){%[1]c        i := %[4]d%[1]c        pgm, _ := os.Executable()%[1]c        dec := strings.HasSuffix(strings.ToLower(pgm), %[2]c/sully%[2]c)%[1]c        if dec == false {i--}%[1]c        c :=%[2]c%[3]s%[2]c%[1]c        src := fmt.Sprintf(%[2]cSully_%%d.go%[2]c, i)%[1]c        exe := fmt.Sprintf(%[2]c./Sully_%%d%[2]c, i)%[1]c        if i<0 {return}%[1]c        f, err := os.Create(src)%[1]c        if err != nil {return}%[1]c        fmt.Fprintf(f, c, 10, 34, c, i)%[1]c        f.Close()%[1]c        if i <= 0 {return}%[1]c        cmd := exec.Command(%[2]cgo%[2]c, %[2]cbuild%[2]c, src)%[1]c        if cmd.Run() != nil {return}%[1]c        exec.Command(exe).Run()}"
        src := fmt.Sprintf("Sully_%d.go", i)
        exe := fmt.Sprintf("./Sully_%d", i)
        if i<0 {return}
        f, err := os.Create(src)
        if err != nil {return}
        fmt.Fprintf(f, c, 10, 34, c, i)
        f.Close()
        if i <= 0 {return}
        cmd := exec.Command("go", "build", src)
        if cmd.Run() != nil {return}
        exec.Command(exe).Run()}